### ISSUE TITLE
PipelineRun Fails After Following Steps Under Trigger Pipeline

### ISSUE THREAD

**Issue by danielhelfand** [openshift/pipelines-tutorial#6]
After running the steps in this tutorial using an RHPDS 4.1 cluster, I triggered a pipeline after creating the git and image `PipelineResources` by running `oc create -f petclinic-deploy-pipelinerun.yaml` with the following:

```yaml
apiVersion: tekton.dev/v1alpha1
kind: PipelineRun
metadata:
  generateName: petclinic-deploy-pipelinerun-
spec:
  pipelineRef:
    name: deploy-pipeline
  trigger:
    type: manual
  serviceAccount: 'pipeline'
  resources:
  - name: app-git
    resourceRef:
      name: petclinic-git
  - name: app-image
    resourceRef:
      name: petclinic-image
```

After running `tkn pr ls` I get:
```
NAME                                 STARTED          DURATION   STATUS   
petclinic-deploy-pipelinerun-4pqkp   15 minutes ago   1 second   Failed  
```

I then looked through the events on my cluster and found the following error message:

```
Missing or invalid Task pipelines-tutorial/s2i-java-8: pods
        "petclinic-deploy-pipelinerun-qwq6k-build-hnq2g-pod-f9e7df" is
        forbidden: [minimum memory usage per Container is 6Mi, but request is
        0., minimum memory usage per Container is 6Mi, but request is 0.,
        minimum memory usage per Container is 6Mi, but request is 0., minimum
        memory usage per Container is 6Mi, but request is 0., minimum memory
        usage per Container is 6Mi, but request is 0., minimum memory usage per
        Container is 6Mi, but request is 0., minimum memory usage per Container
        is 6Mi, but request is 0.]
```

Running `tkn task ls -n pipelines-tutorial` shows that `s2i-java-8` was added:

```
NAME               AGE
openshift-client   1 hour ago
s2i-java-8         1 hour ago
```

**Comment by danielhelfand** [comment]
This could be specific to RHDPS, but it looks like you need to edit the limit range for the `pipelines-tutorial` project to set the limit ranges to a min memory of 0 instead of a min of 6Mi. Why there is a request of 0 as stated above is odd though.

The current workaround, as said in the paragraph above, is to go under the **Administration** tab, select the **Limit Ranges** option, select the `pipelines-tutorial` project, and edit the limit range, as shown below, to set the min values for max/min for memory to 0:

```yaml
kind: LimitRange
apiVersion: v1
metadata:
  name: pipelines-tutorial-core-resource-limits
  namespace: pipelines-tutorial
  selfLink: >-
    /api/v1/namespaces/pipelines-tutorial/limitranges/pipelines-tutorial-core-resource-limits
  uid: 492a15c6-9d90-11e9-ae59-0a8e513d9bd0
  resourceVersion: '1063283'
  creationTimestamp: '2019-07-03T12:44:23Z'
spec:
  limits:
    - type: Container
      max:
        memory: 6Gi
      min:
        memory: '0'
      default:
        cpu: 500m
        memory: 1536Mi
      defaultRequest:
        cpu: 50m
        memory: 256Mi
    - type: Pod
      max:
        memory: 12Gi
      min:
        memory: '0'
```

This allowed the pipeline to run and execute successfully on a 4.1 RHPDS cluster.

**Comment by siamaksade** [comment]
It's probably simpler to just delete the limitrange? 
`oc delete limitrange --all -n pipelines-tutorial`

It would be great if you send the above comments as a PR with the instructions for RHPDS.

**Comment by danielhelfand** [comment]
Deleting the `LimitRange` leads to the following error message:

```
task build failed: resource name may not be empty
```

I verified that the resources as part of the tutorial had been created and that tasks were created as well. Steps to reproduce are below. I ran these after doing `oc create`s to set up everything except the `pipelinerun`.

```
$ oc delete limitrange --all -n pipelines-tutorial
limitrange "pipelines-tutorial-core-resource-limits" deleted

$ oc create -f petclinic-deploy-pipelinerun.yaml 
pipelinerun.tekton.dev/petclinic-deploy-pipelinerun-h7jrn created

$ tkn pr ls
NAME                                 STARTED         DURATION   STATUS   
petclinic-deploy-pipelinerun-h7jrn   4 seconds ago   1 second   Failed   

$ tkn pr logs petclinic-deploy-pipelinerun-h7jrn
task build failed: resource name may not be empty

$tkn taskrun petclinic-deploy-pipelinerun-h7jrn-build-z626b 
NAME                                              STARTED          DURATION     STATUS                   
petclinic-deploy-pipelinerun-h7jrn-build-z626b    10 seconds ago   ---          Failed(CouldntGetTask)   

$tkn taskrun logs petclinic-deploy-pipelinerun-h7jrn-build-z626b
Error: task s2i-java-8 failed: resource name may not be empty 
```

However, lowering the `LimitRange` minimums to 0 allowed for a successful run.

**Comment by gbengataylor** [comment]
The error I received was
```
Failed to create build pod "petclinic-deploy-pipeline-run-l2qgm-build-8tpt5": pods "petclinic-deploy-pipeline-run-l2qgm-build-8tpt5-pod-f3696e" is forbidden: maximum cpu usage per Pod is 2, but limit is 3500m.
```
I updated the LimitRange resource to update the max cpu on the Container for the pipleline to not fail immegiately 

```yaml
kind: LimitRange
apiVersion: v1
metadata:
  name: pipelines-tutorial-core-resource-limits
  namespace: pipelines-tutorial
  selfLink: >-
    /api/v1/namespaces/pipelines-tutorial/limitranges/pipelines-tutorial-core-resource-limits
  uid: 27429129-de34-11e9-98ce-12633bbe5f7a
  resourceVersion: '120638'
  creationTimestamp: '2019-09-23T18:58:38Z'
spec:
  limits:
    - type: Container
      max:
        cpu: '3'
        memory: 6Gi
      min:
        memory: '0'
      default:
        cpu: 500m
        memory: 1536Mi
      defaultRequest:
        cpu: 50m
        memory: 256Mi
    - type: Pod
      max:
        memory: 12Gi
      min:
        memory: '0'
```

**Comment by glamperi** [comment]
Daniel's fix for RHPDS 4.1 is also needed for the RHPDS 4.2.  Thanks Daniel for the workaround!
